diff --git a/deploy_fuel.sh b/deploy_fuel.sh
index dfe5ffe..182590f 100755
--- a/deploy_fuel.sh
+++ b/deploy_fuel.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/bash -x
 
 source functions.sh
 
@@ -15,14 +15,14 @@ fi
 master_name=fuel-master #currently hardcoded in cleanup script
 master_ram=3072
 master_cpu=2
-master_disk=50G
+master_disk=100G
 iso_path=$1
 
 #Cluster nodes
 node_name=fuel-slave #currently hardcoded in cleanup script
-node_ram=3072
-node_cpu=1
-node_size=75G
+node_ram=4096
+node_cpu=2
+node_size=100G
 node_count=$2
 
 #Check and install required packages
@@ -47,3 +47,4 @@ else
 fi
 
 echo "Deployment was finished"
+
diff --git a/deploy_master.sh b/deploy_master.sh
index 2cdd6e7..ebc280a 100755
--- a/deploy_master.sh
+++ b/deploy_master.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/bash -x
 
 source functions.sh
 
diff --git a/deploy_slave.sh b/deploy_slave.sh
index daf4401..602c7a0 100755
--- a/deploy_slave.sh
+++ b/deploy_slave.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/bash -x
 
 source functions.sh
 
@@ -24,7 +24,9 @@ fi
 
 echo "Creating storage..."
 
-virsh vol-create-as --name $name.qcow2 --capacity $size --format qcow2 --allocation $size --pool default
+virsh vol-create-as --name ${name}.qcow2 --capacity $size --format qcow2 --allocation $size --pool default
+virsh vol-create-as --name ${name}2.qcow2 --capacity $size --format qcow2 --allocation $size --pool default
+virsh vol-create-as --name ${name}3.qcow2 --capacity $size --format qcow2 --allocation $size --pool default
 pool_path=$(get_pool_path default)
 
 echo "Starting Fuel slave vm..."
@@ -37,12 +39,14 @@ virt-install \
   --virt-type=kvm \
   --pxe \
   --boot network,hd \
-  --disk "$pool_path/$name.qcow2",cache=writeback,bus=virtio,serial=$(uuidgen) \
+  --disk "$pool_path/${name}.qcow2",cache=writeback,bus=virtio,serial=$(uuidgen) \
+  --disk "$pool_path/${name}2.qcow2",cache=writeback,bus=virtio,serial=$(uuidgen) \
+  --disk "$pool_path/${name}3.qcow2",cache=writeback,bus=virtio,serial=$(uuidgen) \
   --noautoconsole \
   --network network=fuel-pxe,model=$net_driver \
   --network network=$external_network,model=$net_driver \
-  --graphics vnc,listen=0.0.0.0
-#  --cpu host \
+  --graphics vnc,listen=0.0.0.0 \
+#  --cpu host
 #If cpu parameter is set to "host" with QEMU 2.0 hypervisor
 #it causes critical failure during CentOS installation
 
diff --git a/functions.sh b/functions.sh
index e247a0b..f0104f2 100644
--- a/functions.sh
+++ b/functions.sh
@@ -40,7 +40,8 @@ function setup_network {
     gateway_ip=$2
     modprobe nbd max_part=63
     qemu-nbd -n -c /dev/nbd0 $IMAGE_PATH/$name.qcow2
-    sleep 2
+    sleep 5
+    pvscan --cache
     vgscan --mknode
     vgchange -ay os
     sleep 2
diff --git a/manage.sh b/manage.sh
index e9e65f0..15a6ddc 100755
--- a/manage.sh
+++ b/manage.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/bash -x
 
 source functions.sh
 
